### PR TITLE
Update GLM-linear to use numpy.Generator and ArviZ

### PR DESCRIPTION
**Pull request by chiral-carbon**
Addresses issue #81 and aims to make the following changes:

### General Updates
- Uses numpy generator

### ArviZ related
- Uses `figsize`  as kwarg to run `plot_trace` function
 
-----------------------------------------------------------------------------
### Pending
- Checking why the manual model runs slower

### CODE REVIEW

File: examples/generalized_linear_models/GLM-linear.ipynb [hunk 1]
```diff
@@ -103,9 +103,6 @@
    "metadata": {},
```
Comment (OriolAbril) [inline examples/generalized_linear_models/GLM-linear.ipynb:103]: <p>Even though we only use it once, I would keep the rng creation in the 2nd cell plus consequent usage of the <code>rng</code>  object, like the example in https://github.com/pymc-devs/pymc3/wiki/PyMC3-Jupyter-Notebook-Style-Guide</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymc-devs/pymc-examples/pull/149/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (OriolAbril) [inline examples/generalized_linear_models/GLM-linear.ipynb:103]: <p>Can you also add the versions of theano and xarray? I added how to in the last point at https://github.com/pymc-devs/pymc3/wiki/PyMC3-Jupyter-Notebook-Style-Guide</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymc-devs/pymc-examples/pull/149/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='25'/>

**Comment by OriolAbril** [comment]
Looks good, thanks! 

I have two extra general comments in addition to the ones above. The first one is making sure that you have executed the notebook sequentially after editing the code to make sure everything runs correctly, the second is to modify the PR title to something more descriptive _in the context of this repo_, sooner or later all notebooks will be updated for arviz, numpy generator and so on, so finding commits with messages about ArviZ would be difficult, some options that come to mind "update glm-linear to v3 best practices" or "update glm-linear to use numpy.Generator and ArviZ"

File: examples/generalized_linear_models/GLM-linear.ipynb (hunk 1 above)
Comment (chiral-carbon) [reply examples/generalized_linear_models/GLM-linear.ipynb:103]: Sure, I'll add that.

Comment (chiral-carbon) [reply examples/generalized_linear_models/GLM-linear.ipynb:103]: xarray is not being used in this particular notebook so would it be okay to add to it in metadata?

**Comment by chiral-carbon** [comment]
thanks for the suggestions, I'll push these changes. 
Regarding running the notebook, I did run it for testing if the code changes work. But in this particular commit, I have only made the code changes without running the notebook, the reason being that for some reason the manual model seems to run faster for me and the glm.from_formula takes more time. I wanted to ask if you might know what the reason for it could be - is there some dependency that I might be missing or something else because of which I get this result?

**Comment by OriolAbril** [comment]
> the reason being that for some reason the manual model seems to run faster for me and the glm.from_formula takes more time. I wanted to ask if you might know what the reason for it could be - is there some dependency that I might be missing or something else because of which I get this result?

I don't really know, if the performance difference is not systematic you can then ignore the comment on the issue. It is also not worth to go into the source code of the glm module as it will probably be deprecated and replaced by [bambi](https://github.com/bambinos/bambi) (as an external dependency) in `v4`

File: examples/generalized_linear_models/GLM-linear.ipynb (hunk 1 above)
Comment (OriolAbril) [reply examples/generalized_linear_models/GLM-linear.ipynb:103]: It looks like it's not being used but it is being used by ArviZ under the hood, like pymc3 uses theano under the hood even if we don't import or call it explicitly.

**Comment by chiral-carbon** [comment]
Okay, so if I push a new commit with the suggested changes (adding xarray version and keeping `default _rng()` in `cell 2`) should I run the notebook too or not? the notebook will execute, but as I mentioned the output in the model cells would change, and since it may be an isolated issue in my case since we don't know the cause, the output might reflect an inaccurate performance of the model if I run it?

**Comment by OriolAbril** [comment]
Performance will always vary between computers, the runtimes in the docs are at most orientative of the order of magnitude, we should always execute the notebooks and push the changes when modified

**Comment by chiral-carbon** [comment]
understood. thanks for clarifying.

**Comment by chiral-carbon** [comment]
so I tried executing ```%watermark -n -u -v -iv -w (-p theano xarray)``` and it returned a usage error:

```UsageError: unrecognized arguments: (-p theano xarray)```

I tried several things and realized that passing -p PACKAGES with a single package works, but it somehow fails with multiple arguments for me. 

this seemed to work:

```%load_ext watermark

%watermark -n -u -v -iv -p theano 
%watermark -p xarray -w``` 

and it outputs:

```The watermark extension is already loaded. To reload it, use:
  %reload_ext watermark
Last updated: Sat May 01 2021

Python implementation: CPython
Python version       : 3.8.5
IPython version      : 7.13.0

theano: 1.1.2

pymc3     : 3.11.2
numpy     : 1.18.5
arviz     : 0.11.2
matplotlib: 3.3.4

xarray: 0.17.0

Watermark: 2.2.0
```

I wanted your input as to whether you also get this error and if the correction I provided will be acceptable since passing `(-p theano xarray)` is not working in my notebook.

**Comment by OriolAbril** [comment]
I think the error comes from the parenthesis. They are in the guide to indicate that this extra args may not be necessary, many notebooks import theano and xarray explicitly and their versions are listed automatically

**Comment by chiral-carbon** [comment]
okay. running `%watermark -n -u -v -iv -w -p theano xarray` however also returns a usage error, as it doesn't recognize the second arg passed (in this case xarray). The error in this case is: ```UsageError: unrecognized arguments: xarray```.

Should I push the commit by breaking the `%watermark` statement into 2 statements to pass the flags as I have mentioned above? It doesn't align with the style guide but prints the package versions as desired.

**Comment by OriolAbril** [comment]
My bad,  I assumed spaces would work without looking carefully into it. It looks like the separation between theano and xarray has to be a comma instead.

Source: https://github.com/rasbt/watermark/blob/master/watermark/watermark.py#L180

**Comment by chiral-carbon** [comment]
great, got it!

**Review by OriolAbril — APPROVED** [review]

**Pull request merged**